lte/gateway/c/session_manager/SessionProxyResponderHandler.cpp

```diff
@@ -93,10 +101,25 @@ void SessionProxyResponderHandlerImpl::PolicyReAuth(
   enforcer_->get_event_base().runInEventBaseThread([this, request_cpy,
                                                     response_callback]() {
     PolicyReAuthAnswer ans;
-    auto session_map = session_store_.read_sessions({request_cpy.imsi()});
+    const std::string& imsi = request_cpy.imsi();
+    const std::string& session_id = request_cpy.session_id();
+    SessionSearchCriteria criteria(imsi, IMSI_AND_SESSION_ID, session_id);
+    auto session_map = session_store_.read_sessions({imsi});
+    auto session_it = session_store_.find_session(session_map, criteria);
+    auto& session = **session_it;
+
+    const auto& config = session->get_config();
+    const auto& common_context = config.common_context;
+
     SessionUpdate update =
         SessionStore::get_default_session_update(session_map);
-    enforcer_->init_policy_reauth(session_map, request_cpy, ans, update);
+    if (common_context.rat_type() == TGPP_NR) {
+      MLOG(MDEBUG) << "(Policy) ReAuthRequest for 5G from PCF";
+      m5genforcer_->init_policy_reauth(imsi, session_map, request_cpy, ans,
+                                       update);
+    } else {
+      enforcer_->init_policy_reauth(session_map, request_cpy, ans, update);
+    }
```

Comment:
to make it less confussing maybe it would be good to add logs here and remove the log at 123

```
    if (common_context.rat_type() == TGPP_NR) {

      m5genforcer_->init_policy_reauth(imsi, session_map, request_cpy, ans,
                                       update);
      MLOG(MDEBUG) << "Result of 5G N7 (Policy) ReAuthRequest "
                 << raa_result_to_str(ans.result());
    } else {
      enforcer_->init_policy_reauth(session_map, request_cpy, ans, update);
      MLOG(MDEBUG) << "Result of 4G Gx (Policy) ReAuthRequest "
                 << raa_result_to_str(ans.result());
    }
```

Reply:
Maybe you need to modify the log at  line 104 and 135

Reply:
I have modified the logs as suggested, let me know if still something to modify.

Reply:
Maybe now the log `MLOG(MDEBUG) << "Result of` can be outside of the if. This way you save one line